src/Tracking/Conversions.php

```diff
@@ -50,15 +50,15 @@ public function track_event( string $event_name, Data $data ) {
 		$data = $this->prepare_request_data( $event_name, $data );
 
 		try {
-			$this->send_request( $event_name, $data );
-
 			/* translators: 1: Conversions API event name, 2: JSON encoded event data. */
 			$messages = sprintf(
 				'Sending Pinterest Conversions API event %1$s with a payload: %2$s',
 				$event_name,
 				wp_json_encode( $data )
 			);
 			Logger::log( $messages, 'debug', 'conversions' );
+
+			$this->send_request( $event_name, $data );
```

Comment:
Why do we have this send request moved to after the `logging`?

Reply:
I would argue that this is unnecessary. We should always have one log entry for CAPI event.

In case of success a success message, in case of failure the failure message.

With the approach that you are proposing, we get 2 logs for the failure scenario and 1 log for the success scenario.
This also means that you get failure confirmation and don't get success confirmation.